tests/test_pyav.py

```diff
@@ -315,3 +315,36 @@ def test_invalid_resource(tmp_path):
 
     with pytest.raises(IOError):
         iio.imwrite(tmp_path / "foo.abc", img, plugin="pyav")
+
+
+def test_bayer_write():
+    image_shape = (128, 128)
+    image = np.zeros(image_shape, dtype="uint8")
+    buffer = io.BytesIO()
+
+    with iio.imopen(buffer, "w", plugin="pyav", format_hint=".mp4") as file:
+        image[...] = 0
+        for i in range(256):
+            image[::2, ::2] = i
+            file.write(
+                image, is_batch=False, codec="h264", in_pixel_format="bayer_rggb8"
+            )
+
+        image[...] = 0
+        for i in range(256):
+            image[0::2, 1::2] = i
+            image[1::2, 0::2] = i
+            file.write(
+                image, is_batch=False, codec="h264", in_pixel_format="bayer_rggb8"
+            )
+
+        image[...] = 0
+        for i in range(256):
+            image[1::2, 1::2] = i
+            file.write(
+                image, is_batch=False, codec="h264", in_pixel_format="bayer_rggb8"
+            )
+
+    buffer.seek(0)
+    img = iio.imread(buffer, plugin="pyav")
+    assert img.shape == (768, 128, 128, 3)
```

Comment:
I expected this to be (768, 64, 64, 3), since with bayer you specify the rggb pixels individually, right?

Reply:
Good catch. The result takes the bayer image, converts it into the default pixel format for mp4 (I believe its YUV420p), and then converts it from that to RGB while loading.

I'm actually not sure how ffmpeg debayers images, i.e., how it converts from bayer to YUV. Once I find the time, I will do some digging. In the meantime, maybe @hmaarrfk has an insight here?

Reply:
Maybe it does keep the raw resolution, otherwise you'd lose some precision for the green channel ...

Reply:
This is the most annoying things about raw image sensors. I'm glad we are making it easier to people to use them.

A monochrome sensor with 1000 x 1000 pixels has 1 MP of monochrome data.

A color sensor of the same shape, almost all of which use bayer patterns---some permutation of `rggb`---, will spec 1MP. However, you only have 0.25 MP red, 0.25 MP blue, and 0.5 MP green.

Software, or some image processing IC, then interpolates these images into  the original 1000 x 1000 x 3 color space.

Is there some redundant information? Yes.
Can we do better? Probably..... but..... there is so much inertia behind JPEG (H.264 and H.265 all of which use blocks) that there is little you can do.
Eventually, when you compress, the hope is that the redundancy is found too, and the image gets smaller again.

Reply:
Wikipedi has a nice 1-5 about how this is done: https://en.wikipedia.org/wiki/Bayer_filter

Reply:
Thanks for chiming in @hmaarrfk . So if I get it right then a `(128, 128)` bayer image is actually expected to be converted into a `(128, 128)` RGB image due to interpolation? In that case, I guess what we are testing here is correct :)

Reply:
Correct

Reply:
Ok, ready to merge then!